drivers/volume/linstor/linstor.go

```diff
@@ -0,0 +1,412 @@
+	lw := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			return watchlist.List(options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			return watchlist.Watch(options)
+		},
+	}
+	store, controller := cache.NewInformer(lw, &v1.Node{}, resyncPeriod,
+		cache.ResourceEventHandlerFuncs{},
+	)
+	l.store = store
+
+	go controller.Run(l.stopChannel)
+	return nil
+}
+
+func (l *linstor) String() string {
+	logrus.Infof("called: String()")
+	return driverName
+}
+
+func (l *linstor) InspectNode(id string) (*storkvolume.NodeInfo, error) {
+	return nil, &errors.ErrNotSupported{}
+}
+
+func contains(a []string, e string) bool {
+	for _, x := range a {
+		if x == e {
+			return true
+		}
+	}
+	return false
+}
+
+func (l *linstor) InspectVolume(volumeID string) (*storkvolume.Info, error) {
+	logrus.Infof("called: InspectVolume(%s)", volumeID)
+	rd, err := l.cli.ResourceDefinitions.Get(context.TODO(), volumeID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get resource definition: %w", err)
+	}
+
+	resources, err := l.cli.Resources.GetResourceView(context.TODO(), &lclient.ListOpts{
+		Resource: []string{volumeID},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get resources: %w", err)
+	}
+
+	var nodes []string
+	for _, r := range resources {
+		state := r.Volumes[0].State.DiskState
+		if state != "UpToDate" {
+			logrus.Infof("Resource %s is not UpToDate on node %s, skipping (is %s)",
+				r.Name, r.NodeName, state)
+			continue
+		}
+
+		logrus.Infof("Resource %s has flags: %v", r.Name, r.Flags)
+		if contains(r.Flags, lstor.FlagDiskless) {
+			// skip diskless deployments
+			logrus.Infof("Resource %s is diskless on node %s, skipping",
+				r.Name, r.NodeName)
+			continue
+		}
+		nodes = append(nodes, r.NodeName)
+	}
+
+	vd, err := l.cli.ResourceDefinitions.GetVolumeDefinition(context.TODO(), volumeID, 0)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get volume defintion: %w", err)
+	}
+
+	return &storkvolume.Info{
+		VolumeID:   rd.Name,
+		VolumeName: rd.Name,
+		DataNodes:  nodes,
+		Size:       vd.SizeKib / 1048576, // KiB to GiB
+	}, nil
+}
+
+func (l *linstor) getNodeLabels(nodeInfo *storkvolume.NodeInfo) (map[string]string, error) {
+	obj, exists, err := l.store.GetByKey(nodeInfo.SchedulerID)
+	if err != nil {
+		return nil, err
+	} else if !exists {
+		obj, exists, err = l.store.GetByKey(nodeInfo.StorageID)
+		if err != nil {
+			return nil, err
+		} else if !exists {
+			obj, exists, err = l.store.GetByKey(nodeInfo.Hostname)
+			if err != nil {
+				return nil, err
+			} else if !exists {
+				return nil, fmt.Errorf("node %v/%v/%v not found in cache", nodeInfo.StorageID, nodeInfo.SchedulerID, nodeInfo.Hostname)
+			}
+		}
+	}
+	node := obj.(*v1.Node)
+	return node.Labels, nil
+}
+
+func (l *linstor) mapLinstorStatus(n lclient.Node) storkvolume.NodeStatus {
+	logrus.Infof("Node %s is %s", n.Name, n.ConnectionStatus)
```

Comment:
I'll leave it up to you but adding these messages at the start of each function is going to flood the logs and not provide much debug value.